lms/static/js/student_account/views/RegisterView.js

```diff
@@ -77,6 +98,130 @@
+                    return $('#' + $el.attr('id') + '-required-label');
+                },
+
+                renderLiveValidations: function($el, decisions) {
+                    var $label = this.getLabel($el),
+                        $requiredTextLabel = this.getRequiredTextLabel($el),
+                        $icon = this.getIcon($el),
+                        $errorTip = this.getErrorTip($el),
+                        name = $el.attr('name'),
+                        type = $el.attr('type'),
+                        isCheckbox = type === 'checkbox',
+                        hasError = decisions.validation_decisions[name] !== '',
+                        error = isCheckbox ? '' : decisions.validation_decisions[name];
+
+                    if (hasError) {
+                        this.renderLiveValidationError($el, $label, $requiredTextLabel, $icon, $errorTip, error);
+                    } else {
+                        this.renderLiveValidationSuccess($el, $label, $requiredTextLabel, $icon, $errorTip);
+                    }
+                },
+
+                getLabel: function($el) {
+                    return this.$form.find('label[for=' + $el.attr('id') + ']');
+                },
+
+                getIcon: function($el) {
+                    return $('#' + $el.attr('id') + '-validation-icon');
+                },
+
+                getErrorTip: function($el) {
+                    return $('#' + $el.attr('id') + '-validation-error');
+                },
+
+                renderLiveValidationError: function($el, $label, $req, $icon, $tip, error) {
+                    this.removeLiveValidationIndicators(
+                        $el, $label, $req, $icon,
+                        'success', this.positiveValidationIcon
+                    );
+                    this.addLiveValidationIndicators(
+                        $el, $label, $req, $icon, $tip,
+                        'error', this.negativeValidationIcon, error
+                    );
+                    this.renderRequiredMessage($el);
+                },
+
+                renderLiveValidationSuccess: function($el, $label, $req, $icon, $tip) {
+                    var self = this,
+                        name = $el.attr('name'),
+                        validationFadeTime = this.successfulValidationDisplaySeconds * 1000;
+                    this.removeLiveValidationIndicators(
+                        $el, $label, $req, $icon,
+                        'error', this.negativeValidationIcon
+                    );
+                    this.addLiveValidationIndicators(
+                        $el, $label, $req, $icon, $tip,
+                        'success', this.positiveValidationIcon, ''
+                    );
+                    this.hideRequiredMessage($el);
+
+                    // Hide success indicators after some time.
+                    if (this.liveValidationFields[name]) {
+                        clearTimeout(this.liveValidationFields[name]);
```

Comment:
I mean, set it to null inside the function that the timeout executes - so it would only be cleared out when that function executes (at the end of the timeout).